src/app/mat-quill/mat-quill-base.ts

```diff
@@ -0,0 +1,199 @@
+		public ngControl: NgControl,
+		elementRef: ElementRef,
+		domSanitizer: DomSanitizer,
+		doc: any,
+		platformId: any,
+		renderer: Renderer2,
+		zone: NgZone,
+		config: QuillConfig,
+		service: QuillService
+	) {
+		super(
+			elementRef, domSanitizer, doc, platformId,
+			renderer, zone, config, service
+		)
+	}
+}
+
+const _MatQuillMixinBase: CanUpdateErrorStateCtor & CanDisableCtor & typeof MatQuillBase =
+	mixinErrorState(mixinDisabled(MatQuillBase));
+
+@Directive()
+export abstract class _MatQuillBase
+	extends _MatQuillMixinBase
+	implements AfterViewInit, CanDisable, CanUpdateErrorState,
+		ControlValueAccessor, HasErrorState,
+		MatFormFieldControl<any>, OnChanges,
+		OnDestroy, Validator
+{
+	abstract controlType: string;
+	focused = false;
+	abstract id: string;
+
+	constructor(
+		defaultErrorStateMatcher: ErrorStateMatcher,
+		@Optional() parentForm: NgForm,
+		@Optional() parentFormGroup: FormGroupDirective,
+		@Optional() @Self() public ngControl: NgControl,
+		elementRef: ElementRef,
+		domSanitizer: DomSanitizer,
+		@Inject(DOCUMENT) doc: any,
+		@Inject(PLATFORM_ID) platformId: any,
+		renderer: Renderer2,
+		zone: NgZone,
+		@Inject(QUILL_CONFIG_TOKEN) @Optional() config: QuillConfig,
+		service: QuillService
+	) {
+		super(
+			defaultErrorStateMatcher, parentForm, parentFormGroup, ngControl,
+			elementRef, domSanitizer, doc, platformId, renderer, zone, config, service
+		)
+
+		if (this.ngControl != null) {
+			this.ngControl.valueAccessor = this;
+		}
+
+		this.onBlur.subscribe(() => {
+			this.focused = false;
+			this.stateChanges.next();
+		});
+		this.onFocus.subscribe(() => {
+			this.focused = true;
+			this.stateChanges.next();
+		});
+	}
+
+	/*
+	 * GETTERS & SETTERS
+	 */
+
+	@Input()
+	get disabled(): boolean
+	{
+		if (this.ngControl && this.ngControl.disabled !== null) {
+			return this.ngControl.disabled;
+		}
+		return this._disabled;
+	}
+	set disabled(value: boolean)
+	{
+		this._disabled = coerceBooleanProperty(value);
+
+		// Browsers may not fire the blur event if the input is disabled too quickly.
+		// Reset from here to ensure that the element doesn't become stuck.
+		if (this.focused) {
+			this.focused = false;
+			this.stateChanges.next();
+		}
+	}
+	protected _disabled = false;
+
+	get empty() {
+		return coerceBooleanProperty(this.value);
+	}
+
+	@Input()
+	get placeholder(): string { return this._placeholder; }
+	set placeholder(value: string) {
+		this._placeholder = value;
+		this.stateChanges.next();
+	}
+	protected _placeholder: string;
+
+	@Input()
+	get required(): boolean { return this._required; }
+	set required(value: boolean) {
+		this._required = coerceBooleanProperty(value);
+	}
+	protected _required = false;
+
+	@HostBinding('class.floating')
+	get shouldLabelFloat() {
+		return this.focused || !this.empty;
+	}
+
+	get value(): any
+	{
+		try {
+			return this.valueGetter(this.quillEditor, this.editorElem!);
+		} catch(e) {
+			return;
+		}
```

Comment:
`value` is needed immediately upon initialization, whereas `editorElem` is only ready `AfterViewInit`. This try / catch avoids more complicated solutions.